### PR TITLE
uploaded ndvi assignment

### PR DESCRIPTION
**Pull request by fergusca**
# Pull Request Template

## CI Checks
- [x] The notebook runs from start to finish on all operating systems:
   - [x] Mac
   - [x] Windows
   - [x] Linux 
### Reproducibility
- [x] Are the data downloaded in the code
- [x] Are paths created to ensure they work on all operating systems (using  os.path.join)
- [x] Are comments used to clarify the contents of the code that can’t be clarified using expressive variable and function names alone? (not too many comments - just enough)
- [x] Does the notebook run from start to finish?

### Functions
- [x] Do functions follow PEP 8 format conventions?
- [x] Are function docstrings clear (all  inputs and outputs clearly  described and defined)
- [x] Are function names expressive (the name describes what the function does)?
- [x] Are functions easy to understand and read?
- [x] How many tasks does each function do? (ideally a function does one thing well).
### Package imports
- [x] Are standard modules (those included with the base  python install) vs. third party (related but externally developed tools) import groups correct with appropriate spacing in between each group?
- [x] Are variable names throughout the code, expressive?
- [x] Suggest changes if not, highlight what was done well
- [x] Is the code overall easy to understand and read? Are there things that would make it more clear / cleaner?

### DRY Code
- [x] Are segments of code repeated in the file or is the code DRY?
- [x] Are loops used to optimize DRY code?
- [x] Are functions used to optimize DRY code?
- [x] Are there any areas that could be potentially improved (you can suggest improvements OR you can just highlight parts of the code where you suspect it could  be cleaner / more efficient.

### Novel Approaches to Problem solving
- [x] Highlight any novel approaches to completing the assignment.

### CODE REVIEW

File: fergus_emi_ea-2021-04-ndvi-automation.py [hunk 1]
```diff
@@ -0,0 +1,758 @@
...
+    ndvi_xr = (all_bands[1]-all_bands[0]) / (all_bands[1]+all_bands[0])
+
+    # Apply cloud mask to NDVI
+    ndvi_mask = ndvi_xr.where(~pixel_qa.isin(vals))
+
+    return ndvi_mask
+
+
+###############
+#  FUNCTION to create a pandas dataframe from NDVI and grab site and date from file path
```
Comment (krantz23) [inline fergus_emi_ea-2021-04-ndvi-automation.py:321]: This function could be placed at the top of the notebook with your other functions

**Comment by fergusca** [comment]
I reviewed Korrina Rantz's code - I wasn't able to edit the checklist on her conversation tab.

File: fergus_emi_ea-2021-04-ndvi-automation.py [hunk 2]
```diff
@@ -0,0 +1,758 @@
...
+
+    return mean_ndvi_df
+
+
+# Note: "from_disk" needs updated rioxarray package
+
+
+# In[4]:
+
+
```
Comment (krantz23) [inline fergus_emi_ea-2021-04-ndvi-automation.py:372]: These comments and hints can be cleaned up once you complete the task

**Review by krantz23 — COMMENTED** [review]
@fergusca I think you did a great job on this and had creative ways to find solutions. Sorry that my coding isn't better and I can't provide much insight on how to automate things more efficiently.

File: fergus_emi_ea-2021-04-ndvi-automation.py (hunk 2 above)
Comment (fergusca) [reply fergus_emi_ea-2021-04-ndvi-automation.py:372]: Thanks, Korrina for the helpful feedback! I agree that resampling by month was not necessary and may explain the weird x-axis in the plot. It's something I should clean up down the road. And I appreciate you catching the extra libraries that were not needed to run the notebook. Thanks for reviewing my code!

**Pull request closed without merging**